.azurePipeline/templatePublishLogsFromPods.yml

```diff
@@ -0,0 +1,30 @@
+# Template to login into Dockerhub, build a docker image and push it.
+
+# Parameters:
+# `logsFolderPath` is the folder where the logs from the pods will be stored.
+# `releaseName` is the name of the release used to get the relevant list of pods.
+# `kubeConfigFile` is the kubeConfigFile path.
+
+parameters:
+  logsFolderPath: ''
+  releaseName: ''
+  kubeConfigFile: ''
+
+steps:
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '3.7'
+  condition: failed()
+- script: |
+    echo "Copy the pods' logs from the release '${{ parameters.releaseName }}' to '${{ parameters.logsFolderPath }}'."
+    python .azurePipeline/getPodsLogs.py ${{ parameters.logsFolderPath }} ${{ parameters.releaseName }}
+  displayName: 'Copy the logs from the service pods.'
+  condition: failed()
```

Comment:
Do we have to repeat the condition in this template, can we lift it up to the calling location in `azure-pipelines.yml`?

Reply:
The step `template` does not accept a `condition` field, which is why I needed to bring them back here.